app/javascript/widget/views/Home.vue

```diff
@@ -1,26 +1,72 @@
+    v-if="!conversationSize && isFetchingList"
+    class="flex flex-1 items-center h-full bg-black-25 justify-center"
+  >
+    <spinner size=""></spinner>
+  </div>
+  <div v-else class="home">
     <div class="header-wrap">
-      <ChatHeaderExpanded
-        v-if="isHeaderExpanded && !hideWelcomeHeader"
-        :intro-heading="introHeading"
-        :intro-body="introBody"
-        :avatar-url="channelConfig.avatarUrl"
-        :show-popout-button="showPopoutButton"
-      />
-      <ChatHeader
-        v-else
-        :title="channelConfig.websiteName"
-        :avatar-url="channelConfig.avatarUrl"
-        :show-popout-button="showPopoutButton"
-      />
+      <transition
+        enter-active-class="transition-all delay-200 duration-300 ease-out"
+        leave-active-class="transition-all duration-200 ease-in"
+        enter-class="opacity-0 transform -translate-y-16"
+        enter-to-class="opacity-100 transform translate-y-0"
+        leave-class="opacity-100 transform translate-y-0"
+        leave-to-class="opacity-0 transform -translate-y-16"
+      >
+        <chat-header-expanded
+          v-if="!isOnMessageView"
+          :intro-heading="introHeading"
+          :intro-body="introBody"
+          :avatar-url="channelConfig.avatarUrl"
+          :show-popout-button="showPopoutButton"
+        />
+      </transition>
+      <transition
+        enter-active-class="transition-all delay-200 duration-300 ease-out"
+        leave-active-class="transition-all duration-200 ease-in"
+        enter-class="opacity-0 transform -translate-y-16"
+        enter-to-class="opacity-100 transform translate-y-0"
+        leave-class="opacity-100 transform translate-y-0"
+        leave-to-class="opacity-0 transform -translate-y-16"
+      >
+        <chat-header
+          v-if="isOnMessageView"
+          :title="channelConfig.websiteName"
+          :avatar-url="channelConfig.avatarUrl"
+          :show-popout-button="showPopoutButton"
+        />
+      </transition>
     </div>
-    <AvailableAgents v-if="showAvailableAgents" :agents="availableAgents" />
-    <ConversationWrap :grouped-messages="groupedMessages" />
+    <conversation-wrap :grouped-messages="groupedMessages" />
     <div class="footer-wrap">
-      <div v-if="showInputTextArea" class="input-wrap">
-        <ChatFooter />
-      </div>
+      <transition
+        enter-active-class="transition-all delay-300 duration-300 ease-out"
+        leave-active-class="transition-all duration-200 ease-in"
+        enter-class="opacity-0 transform translate-y-16"
+        enter-to-class="opacity-100 transform translate-y-0"
+        leave-class="opacity-100 transform translate-y-0"
+        leave-to-class="opacity-0 transform translate-y-16"
+      >
+        <div v-if="showInputTextArea && isOnMessageView" class="input-wrap">
+          <chat-footer />
+        </div>
+      </transition>
+      <transition
+        enter-active-class="transition-all delay-300 duration-300 ease-out"
```

Comment:
Remove translation-all and target only the affecting properties, Like `transition-opacity transition-transform` if they are available, Also see if we can add `will-change` css property to these with `transform, opacity` to get butter smooth animations.

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/will-change

Reply:
I guess will-change needs to be added only if there are problems with transition performance. Abusing it is not recommended.

I will change the transition-all to transform and opacity